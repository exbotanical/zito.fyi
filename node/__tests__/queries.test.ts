import {
  allPostsByCategoryQuery,
  allPostsByTagQuery,
  allPostsQuery,
} from '../../src/templates/feed/queries'
import { postsListQueryResponse } from '../../test/fixtures'
import {
  getAllPostsByCategory,
  getAllPostsByTag,
  getAllPosts,
} from '../queries'

jest.spyOn(global.console, 'warn').mockImplementation()
jest.spyOn(global.console, 'error').mockImplementation()

const categoryQueryResponse = {
  data: {
    allMdx: {
      edges: postsListQueryResponse.allMdx.edges.filter(
        edge => edge.node.frontmatter?.category === 'category',
      ),
    },
  },
}
const tagQueryResponse = {
  data: {
    allMdx: {
      edges: postsListQueryResponse.allMdx.edges.filter(edge =>
        edge.node.frontmatter?.tags?.includes('tag'),
      ),
    },
  },
}
const indexQueryResponse = { data: postsListQueryResponse }

describe('build util `getAllPosts`', () => {
  it('returns the index feed', async () => {
    const graphql = jest.fn().mockResolvedValue(indexQueryResponse)

    const posts = await getAllPosts(graphql)

    expect(graphql).toHaveBeenCalledWith(
      allPostsQuery,
      expect.objectContaining({ today: expect.any(Date) }),
    )

    expect(posts).toMatchSnapshot()
  })

  it('warns when no data is returned', async () => {
    const graphql = jest.fn().mockResolvedValue({ data: undefined })

    const posts = await getAllPosts(graphql)

    expect(console.warn).toHaveBeenCalled()
    expect(posts).toStrictEqual([])
  })

  it('propagates query errors', async () => {
    const graphql = jest.fn().mockResolvedValue({ errors: 'TEST ERROR' })

    await expect(getAllPosts(graphql)).rejects.toThrow('TEST ERROR')

    expect(console.error).toHaveBeenCalled()
    expect(console.error).toHaveBeenCalledWith('TEST ERROR')
  })
})

describe('build util `getAllPostsByTag`', () => {
  it('returns the tag feed', async () => {
    const graphql = jest.fn().mockResolvedValue(tagQueryResponse)

    const posts = await getAllPostsByTag(graphql, 'tag')

    expect(graphql).toHaveBeenCalledWith(allPostsByTagQuery, {
      tag: 'tag',
    })
    expect(posts).toMatchSnapshot()
  })
})

describe('build util `getAllPostsByCategory`', () => {
  it('returns the category feed', async () => {
    const graphql = jest.fn().mockResolvedValue(categoryQueryResponse)

    const posts = await getAllPostsByCategory(graphql, 'category')

    expect(graphql).toHaveBeenCalledWith(allPostsByCategoryQuery, {
      category: 'category',
    })

    expect(posts).toMatchSnapshot()
  })
})
