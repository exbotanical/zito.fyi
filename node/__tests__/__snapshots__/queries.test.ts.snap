// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`build util \`getAllPosts\` returns the index feed 1`] = `
[
  {
    "body": undefined,
    "category": "technology",
    "contentFilePath": undefined,
    "coverImageAlt": "An image.",
    "coverImageUrl": undefined,
    "coverImg": {
      "backgroundColor": "transparent",
      "height": 368,
      "images": {
        "fallback": {
          "sizes": "(min-width: 920px) 920px, 100vw",
          "src": "/static/351fe0crb13d2648e4bbd3fe9tae3gf4/df323/de.jpg",
          "srcSet": "/static/351fe0crb13d2648e4bbd3fe9tae3gf4/2fb23/de.jpg 230w,
/static/351fe0crb13d2648e4bbd3fe9tae3gf4/1fc23/de.jpg 460w,
/static/351fe0crb13d2648e4bbd3fe9tae3gf4/df323/de.jpg 920w,
/static/351fe0crb13d2648e4bbd3fe9tae3gf4/3f823/de.jpg 1840w",
        },
        "sources": [
          {
            "sizes": "(min-width: 920px) 920px, 100vw",
            "srcSet": "/static/351fe0crb13d2648e4bbd3fe9tae3gf4/ef223/de.avif 230w,
/static/351fe0crb13d2648e4bbd3fe9tae3gf4/5f723/de.avif 460w,
/static/351fe0crb13d2648e4bbd3fe9tae3gf4/9f123/de.avif 920w,
/static/351fe0crb13d2648e4bbd3fe9tae3gf4/bfe23/de.avif 1840w",
            "type": "image/avif",
          },
          {
            "sizes": "(min-width: 920px) 920px, 100vw",
            "srcSet": "/static/351fe0crb13d2648e4bbd3fe9tae3gf4/1f623/de.webp 230w,
/static/351fe0crb13d2648e4bbd3fe9tae3gf4/ffb23/de.webp 460w,
/static/351fe0crb13d2648e4bbd3fe9tae3gf4/df423/de.webp 920w,
/static/351fe0crb13d2648e4bbd3fe9tae3gf4/3f223/de.webp 1840w",
            "type": "image/webp",
          },
        ],
      },
      "layout": "constrained",
      "placeholder": {
        "fallback": "data:image/jpeg;base64,/9j/2wBDABALDA4MChAODQ4SERATGCgaGBYWGDEjJR0oOjM9PDkzODdASFxOQERXRTc4UG1RV19iZ2hnPk1xeXBkeFxlZ2P/2wBDARESEhgVGC8aGi9jQjhCY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2P/wgARCAAIABQDASIAAhEBAxEB/8QAFgABAQEAAAAAAAAAAAAAAAAAAAMF/8QAFQEBAQAAAAAAAAAAAAAAAAAAAQD/2gAMAwEAAhADEAAAAdGYUwP/xAAXEAEBAQEAAAAAAAAAAAAAAAAAEQIT/9oACAEBAAEFAqjrp//EABURAQEAAAAAAAAAAAAAAAAAAAEQ/9oACAEDAQE/ASf/xAAUEQEAAAAAAAAAAAAAAAAAAAAQ/9oACAECAQE/AT//xAAaEAACAgMAAAAAAAAAAAAAAAAAAQIRInHh/9oACAEBAAY/AqR0xg62f//EABkQAAIDAQAAAAAAAAAAAAAAAAERACFBgf/aAAgBAQABPyFEhrNlmSSyKigdBJ//2gAMAwEAAgADAAAAEPf/AP/EABURAQEAAAAAAAAAAAAAAAAAAAEQ/9oACAEDAQE/EAE//8QAFhEBAQEAAAAAAAAAAAAAAAAAARAR/9oACAECAQE/EF2f/8QAGhABAAIDAQAAAAAAAAAAAAAAAQARITFBkf/aAAgBAQABPxCtWQ3cIhEGOxxx4GpTYkWILyf/2Q==",
      },
      "width": 920,
    },
    "dateModified": 2020-04-11T00:00:00.000Z,
    "datePublished": 2020-04-11T00:00:00.000Z,
    "description": "Description",
    "internalContent": undefined,
    "pathName": "/test-one",
    "route": "/test-one",
    "slug": "/test-one",
    "tags": [
      "programming",
      "cate",
      "ee",
    ],
    "timeToRead": "1 min read",
    "title": "Test One",
    "url": "https://zito.fyi/test-one",
  },
  {
    "body": undefined,
    "category": "technology",
    "contentFilePath": undefined,
    "coverImageAlt": "An image",
    "coverImageUrl": undefined,
    "coverImg": {
      "backgroundColor": "transparent",
      "height": 368,
      "images": {
        "fallback": {
          "sizes": "(min-width: 920px) 920px, 100vw",
          "src": "/static/351fe0crb13d2648e4bbd3fe9tae3gf4/df323/de.jpg",
          "srcSet": "/static/351fe0crb13d2648e4bbd3fe9tae3gf4/2fb23/de.jpg 230w,
/static/351fe0crb13d2648e4bbd3fe9tae3gf4/1fc23/de.jpg 460w,
/static/351fe0crb13d2648e4bbd3fe9tae3gf4/df323/de.jpg 920w,
/static/351fe0crb13d2648e4bbd3fe9tae3gf4/3f823/de.jpg 1840w",
        },
        "sources": [
          {
            "sizes": "(min-width: 920px) 920px, 100vw",
            "srcSet": "/static/351fe0crb13d2648e4bbd3fe9tae3gf4/ef223/de.avif 230w,
/static/351fe0crb13d2648e4bbd3fe9tae3gf4/5f723/de.avif 460w,
/static/351fe0crb13d2648e4bbd3fe9tae3gf4/9f123/de.avif 920w,
/static/351fe0crb13d2648e4bbd3fe9tae3gf4/bfe23/de.avif 1840w",
            "type": "image/avif",
          },
          {
            "sizes": "(min-width: 920px) 920px, 100vw",
            "srcSet": "/static/351fe0crb13d2648e4bbd3fe9tae3gf4/1f623/de.webp 230w,
/static/351fe0crb13d2648e4bbd3fe9tae3gf4/ffb23/de.webp 460w,
/static/351fe0crb13d2648e4bbd3fe9tae3gf4/df423/de.webp 920w,
/static/351fe0crb13d2648e4bbd3fe9tae3gf4/3f223/de.webp 1840w",
            "type": "image/webp",
          },
        ],
      },
      "layout": "constrained",
      "placeholder": {
        "fallback": "data:image/jpeg;base64,/9j/2wBDABALDA4MChAODQ4SERATGCgaGBYWGDEjJR0oOjM9PDkzODdASFxOQERXRTc4UG1RV19iZ2hnPk1xeXBkeFxlZ2P/2wBDARESEhgVGC8aGi9jQjhCY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2P/wgARCAAIABQDASIAAhEBAxEB/8QAFgABAQEAAAAAAAAAAAAAAAAAAAMF/8QAFQEBAQAAAAAAAAAAAAAAAAAAAQD/2gAMAwEAAhADEAAAAdGYUwP/xAAXEAEBAQEAAAAAAAAAAAAAAAAAEQIT/9oACAEBAAEFAqjrp//EABURAQEAAAAAAAAAAAAAAAAAAAEQ/9oACAEDAQE/ASf/xAAUEQEAAAAAAAAAAAAAAAAAAAAQ/9oACAECAQE/AT//xAAaEAACAgMAAAAAAAAAAAAAAAAAAQIRInHh/9oACAEBAAY/AqR0xg62f//EABkQAAIDAQAAAAAAAAAAAAAAAAERACFBgf/aAAgBAQABPyFEhrNlmSSyKigdBJ//2gAMAwEAAgADAAAAEPf/AP/EABURAQEAAAAAAAAAAAAAAAAAAAEQ/9oACAEDAQE/EAE//8QAFhEBAQEAAAAAAAAAAAAAAAAAARAR/9oACAECAQE/EF2f/8QAGhABAAIDAQAAAAAAAAAAAAAAAQARITFBkf/aAAgBAQABPxCtWQ3cIhEGOxxx4GpTYkWILyf/2Q==",
      },
      "width": 920,
    },
    "dateModified": 2020-04-11T00:00:00.000Z,
    "datePublished": 2020-04-11T00:00:00.000Z,
    "description": "Description",
    "internalContent": undefined,
    "pathName": "/test-ones",
    "route": "/test-ones",
    "slug": "/test-ones",
    "tags": [
      "programming",
      "writing",
    ],
    "timeToRead": "1 min read",
    "title": "Test Ones",
    "url": "https://zito.fyi/test-ones",
  },
]
`;

exports[`build util \`getAllPostsByCategory\` returns the category feed 1`] = `[]`;

exports[`build util \`getAllPostsByTag\` returns the tag feed 1`] = `[]`;
